src/gui/SubWindow.cpp

```diff
@@ -220,22 +235,34 @@ void SubWindow::resizeEvent( QResizeEvent * event )
 	{
 		buttonBarWidth = buttonBarWidth + m_buttonSize.width() + buttonGap;
 		m_maximizeBtn->move( middleButtonPos );
+	// on Mac the isMaximize() function returns always false
```

Comment:
1. I'm unsure if I understand you right here. Qt::WindowMaximizeButtonHint is a Flag which returns true if the window is able to maximize. For example the Fx-Mixer window is not able to be maximized therefore this flag returns false. And if it's false we don't need the maximize button.

2. We could implement the isMaximized() function in this form:
```

bool SubWindow::isMaximized()
{
#ifdef LMMS_BUILD_APPLE
	// check if subwindow size is identical to the MdiArea size, accounting for scrollbars
	int hScrollBarHeight = mdiArea()->horizontalScrollBar()->isVisible() ? mdiArea()->horizontalScrollBar()->size().height() : 0;
	int vScrollBarWidth = mdiArea()->verticalScrollBar()->isVisible() ? mdiArea()->verticalScrollBar()->size().width() : 0;
	QSize areaSize( this->mdiArea()->size().width() - vScrollBarWidth, this->mdiArea()->size().height() - hScrollBarHeight );

	return areaSize == this->size();
#else
	return QMdiSubWindow::isMaximized();
#endif
}

```
So we have one function for all systems. But for what I see at the first look, we have on Linux & Co. one more redundant function call. I'll look tomorrow into it. 

Regards